### ISSUE TITLE
Links in the infobox are not modified by hostname_replace

### ISSUE THREAD

**Issue by return42** [searxng/searxng#1348]
The `hostname_replace` list is not applied to the URLs of links and images in the infobox:

- `infobox.img_src`
https://github.com/searxng/searxng/blob/8f7cb4ad6fa3f9dec4b89807a32fa69e0e72bb84/searx/templates/simple/infobox.html#L3

- `for attribute in infobox.attributes` --> `attribute.image.src`
https://github.com/searxng/searxng/blob/8f7cb4ad6fa3f9dec4b89807a32fa69e0e72bb84/searx/templates/simple/infobox.html#L7-L11

- `for url in infobox.urls` --> `url.url`
https://github.com/searxng/searxng/blob/8f7cb4ad6fa3f9dec4b89807a32fa69e0e72bb84/searx/templates/simple/infobox.html#L23-L25

IDK if the above is a complete list of all URLs in the infobox / maybe I have overseen some URLs.

**Comment by dalf** [comment]
With the current architecture, the plugin must parse the infobox results.

---

if the plugin returns `False` then the infobox is removed:
https://github.com/searxng/searxng/blob/8f7cb4ad6fa3f9dec4b89807a32fa69e0e72bb84/searx/results.py#L205-L206

**Comment by return42** [comment]
> With the current architecture, the plugin must parse the infobox results.

Yes, but its an complex data type that is documented not yet .. we can reverse engineer the type from the usage (see. https://github.com/searxng/searxng/issues/1348#issue-1275816442 ) in the infobox template.

And we have this complex `merge_two_infoboxes()` function that can give some additional hints.

https://github.com/searxng/searxng/blob/8f7cb4ad6fa3f9dec4b89807a32fa69e0e72bb84/searx/results.py#L58

That reminds me, that we clearly document our internal API.

Maybe I will have a look about the next days.

**Comment by return42** [comment]
> Maybe I will have a look about the next days.

I sent PR #1412 to cover this topic well.
